### ISSUE TITLE
Workflow 1-2: Birthdate generated is an invalid date

### ISSUE THREAD

**Issue by aleksa-krolls** [OpenFn/openfn-lime#4]
## Describe the bug and expected behavior 
See issue #1 for background. 

The custom function `calculateDOB()` sometimes generates invalidate date values like `"birthdate": "2016-4-31"` for OpenMRS patients we are trying to create. April 31 is not a valid date.
https://github.com/OpenFn/openfn-lime/blob/main/jobs/wf1-2-createPatients.js#L43-L51

See example run failure with error message `Value 31 for dayOfMonth must be in the range [1,30]]`:  
https://www.openfn.org/projects/msf-lime/runs/06476de9-e977-7a90-b5a8-093d46d6bae7

## To Reproduce
1. Create a new branch off of `main` 
(NOTE: I had to make edits to your code so it would work as a flow job - e.g., change FROM `const { trackedEntityInstances } = state;` TO `const { trackedEntityInstances } = state.data;`). 
2. Comment back in L81: https://github.com/OpenFn/openfn-lime/blob/main/jobs/wf1-2-createPatients.js#L80-L81
3. Re-run the workflow to reproduce the error in job `wf1-2` 

## Request
1. Please ensure `birthdate` returns a valid date (e.g., `"YYYY-MM-ddT00:00:00.000+0000"`)
2. also add the mapping `birthdateEstimated: true`

Example output if dhis2 `age` = 23: 
```
"age": 23,
"birthdate": "2000-05-31T00:00:00.000+0000",
"birthdateEstimated": true,
```

### expression.js
Link to the job itself in Github: https://github.com/OpenFn/openfn-lime/blob/main/jobs/wf1-2-createPatients.js#L80-L81
Adaptor: openmrs@1.0.0

### state.json
1. Run job `wf1-1` to generate state for job 2
See LP for `MSF DHIS2 Test Org (api user)`

3. Run job `wf1-2` to reproduce the error 
See LP for `MSF OpenMRS Demo`

## To test/resolve
1. After the desired output is working locally (from the CLI), please open a pull request.

## Toggl 
`MSF LIME Prototype 2023`

**Comment by aleksa-krolls** [comment]
Abandoned per the client's request.

**Comment by mtuchi** [comment]
@aleksa-krolls you can QA this in the AM as well, See the fix in the PR #6
